crates/matrix-sdk-base/src/room/room_info.rs

```diff
@@ -525,6 +537,9 @@ impl RoomInfo {
 
     /// Set the membership RoomState of this Room
     pub fn set_state(&mut self, room_state: RoomState) {
+        // Changing our state removes the invite details since we can't know that they
```

Comment:
Do you think we should set `invite_acceptance_details` to `None` if and only if the previous state was `RoomState::Invited`? At least it would be make things clearer, and it wouldn't hide a possible bug if any. Like, we may want to throw an (log) error if the previous state is not `Invited` and if `invite_acceptance_details` is `Some`?

Reply:
Hmm, I guess we can add a log for that case.

We also have some tests that confirm that it isn't set when it shouldn't be.